File: app/src/main/java/com/kickstarter/ui/activities/ProjectSocialActivity.java
package com.kickstarter.ui.activities;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;

import com.kickstarter.R;
import com.kickstarter.libs.BaseActivity;
import com.kickstarter.models.Project;
import com.kickstarter.ui.adapters.ProjectSocialAdapter;
import com.kickstarter.ui.viewholders.ProjectContextViewHolder;

import butterknife.Bind;
import butterknife.ButterKnife;
import butterknife.OnClick;

public final class ProjectSocialActivity extends BaseActivity implements ProjectSocialAdapter.Delegate {
  protected @Bind(R.id.project_social_recycler_view) RecyclerView recyclerView;

  @Override
  protected void onCreate(final @Nullable Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.project_social_layout);
    ButterKnife.bind(this);

    final Intent intent = getIntent();
    final Project project = intent.getParcelableExtra(getString(R.string.intent_project));
    final ProjectSocialAdapter adapter = new ProjectSocialAdapter(this);

    recyclerView.setAdapter(adapter);
    recyclerView.setLayoutManager(new LinearLayoutManager(this));
    adapter.takeProject(project);
  }

  @Override
  @OnClick(R.id.back_button)
  public void onBackPressed() {
    super.onBackPressed();
    overridePendingTransition(R.anim.fade_in_slide_in_left, R.anim.slide_out_right);
  }

  @Override
  public void projectContextClicked(final @NonNull ProjectContextViewHolder viewHolder) {
    onBackPressed();
  }
}

File: app/src/main/java/com/kickstarter/ui/activities/ProjectSocialActivity.kt
package com.kickstarter.ui.activities

import android.os.Bundle
import androidx.activity.addCallback
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import com.kickstarter.databinding.ProjectSocialLayoutBinding
import com.kickstarter.libs.utils.extensions.addToDisposable
import com.kickstarter.libs.utils.extensions.getEnvironment
import com.kickstarter.ui.adapters.ProjectSocialAdapter
import com.kickstarter.ui.extensions.finishWithAnimation
import com.kickstarter.utils.WindowInsetsUtil
import com.kickstarter.viewmodels.ProjectSocialViewModel.Factory
import com.kickstarter.viewmodels.ProjectSocialViewModel.ProjectSocialViewModel
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.disposables.CompositeDisposable

class ProjectSocialActivity : AppCompatActivity() {

    private lateinit var binding: ProjectSocialLayoutBinding

    private lateinit var viewModelFactory: Factory
    private val viewModel: ProjectSocialViewModel by viewModels { viewModelFactory }

    private var disposables = CompositeDisposable()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        this.getEnvironment()?.let { env ->
            viewModelFactory = Factory(env, intent = intent)
        }

        binding = ProjectSocialLayoutBinding.inflate(layoutInflater)
        WindowInsetsUtil.manageEdgeToEdge(
            window,
            binding.root
        )
        setContentView(binding.root)

        val adapter = ProjectSocialAdapter()
        binding.projectSocialRecyclerView.adapter = adapter
        binding.projectSocialRecyclerView.layoutManager = LinearLayoutManager(this)

        viewModel.outputs.project()
            .observeOn(AndroidSchedulers.mainThread())
            .subscribe { adapter.takeProject(it) }
            .addToDisposable(disposables)

        this.onBackPressedDispatcher.addCallback {
            finishWithAnimation()
        }
    }

    override fun onDestroy() {
        disposables.clear()
        binding.projectSocialRecyclerView.adapter = null
        super.onDestroy()
    }
}
